source/common/network/utility.cc

```diff
@@ -208,6 +208,47 @@ Address::InstanceConstSharedPtr Utility::getLocalAddress(const Address::IpVersio
   return ret;
 }
 
+bool Utility::isLocalConnection(const Network::ConnectionSocket& socket) {
+
+  const auto& remote_address = socket.remoteAddress();
+  if (remote_address->type() == Envoy::Network::Address::Type::Pipe ||
+      isLoopbackAddress(*remote_address)) {
+    return true;
+  }
+
+  struct ifaddrs* ifaddr;
+  struct ifaddrs* ifa;
```

Comment:
`ifa` can be declared within the scope of the for loop.

Reply:
Done. Thanks.